src/parser.rs

```diff
@@ -3438,173 +3438,199 @@ impl<'a> Parser<'a> {
+                    return self.expected(
+                        "ADD, RENAME, PARTITION or DROP after ALTER TABLE",
+                        self.peek_token(),
+                    );
+                };
+                Ok(Statement::AlterTable {
+                    name: table_name,
+                    operation,
+                })
             }
-        } else if self.parse_keyword(Keyword::PARTITION) {
-            self.expect_token(&Token::LParen)?;
-            let before = self.parse_comma_separated(Parser::parse_expr)?;
-            self.expect_token(&Token::RParen)?;
-            self.expect_keyword(Keyword::RENAME)?;
-            self.expect_keywords(&[Keyword::TO, Keyword::PARTITION])?;
-            self.expect_token(&Token::LParen)?;
-            let renames = self.parse_comma_separated(Parser::parse_expr)?;
-            self.expect_token(&Token::RParen)?;
-            AlterTableOperation::RenamePartitions {
-                old_partitions: before,
-                new_partitions: renames,
-            }
-        } else if self.parse_keyword(Keyword::CHANGE) {
-            let _ = self.parse_keyword(Keyword::COLUMN);
-            let old_name = self.parse_identifier()?;
-            let new_name = self.parse_identifier()?;
-            let data_type = self.parse_data_type()?;
-            let mut options = vec![];
-            while let Some(option) = self.parse_optional_column_option()? {
-                options.push(option);
-            }
-
-            AlterTableOperation::ChangeColumn {
-                old_name,
-                new_name,
-                data_type,
-                options,
-            }
-        } else if self.parse_keyword(Keyword::ALTER) {
-            let _ = self.parse_keyword(Keyword::COLUMN);
-            let column_name = self.parse_identifier()?;
-            let is_postgresql = dialect_of!(self is PostgreSqlDialect);
-
-            let op = if self.parse_keywords(&[Keyword::SET, Keyword::NOT, Keyword::NULL]) {
-                AlterColumnOperation::SetNotNull {}
-            } else if self.parse_keywords(&[Keyword::DROP, Keyword::NOT, Keyword::NULL]) {
-                AlterColumnOperation::DropNotNull {}
-            } else if self.parse_keywords(&[Keyword::SET, Keyword::DEFAULT]) {
-                AlterColumnOperation::SetDefault {
-                    value: self.parse_expr()?,
-                }
-            } else if self.parse_keywords(&[Keyword::DROP, Keyword::DEFAULT]) {
-                AlterColumnOperation::DropDefault {}
-            } else if self.parse_keywords(&[Keyword::SET, Keyword::DATA, Keyword::TYPE])
-                || (is_postgresql && self.parse_keyword(Keyword::TYPE))
-            {
-                let data_type = self.parse_data_type()?;
-                let using = if is_postgresql && self.parse_keyword(Keyword::USING) {
-                    Some(self.parse_expr()?)
+            Keyword::INDEX => {
+                let _ = self.parse_keyword(Keyword::ONLY);
```

Comment:
I think this is ignoring the error -- it should be something like
```suggestion
                self.parse_keyword(Keyword::ONLY)?;
```

However, it isn't clear to me why it is trying to parse an `ONLY` as part of an alter index statement

Reply:
I see this is the same in some statements above. I'll clean it up in a follow on PR

Reply:
Removed in https://github.com/sqlparser-rs/sqlparser-rs/pull/773